csharp/test/TestCommon/TestHelper.cs

```diff
@@ -26,6 +26,50 @@ public interface PlatformAdapter
 
     public abstract class TestHelper
     {
+        // A custom trace listener that always aborts the application upon failure.
+        internal class TestTraceListener : DefaultTraceListener
+        {
+            public override void Fail(string message)
+            {
+                Fail(message, null);
+            }
+
+            public override void Fail(string message, string detailMessage)
+            {
+                StringBuilder sb = new StringBuilder();
+                sb.Append("failed:");
+                if (message != null && message.Length > 0)
+                {
+                    sb.Append(" ");
+                    sb.Append(message);
+                }
+                sb.Append("\n");
+                if (detailMessage != null && detailMessage.Length > 0)
+                {
+                    sb.Append("details:").Append(detailMessage).Append("\n");
```

Comment:
It would be good if the printing of "message" and "detailMessage" was handled the same way. We're missing a space here for instance after the column.